### ISSUE TITLE
didReceiveAttrs not called anymore

### ISSUE THREAD

**Issue by patric-eberle** [mitchlloyd/ember-islands#58]
Ember 3.13.1
ember-islands 1.5.1

It seems like the initial handling of `didReceiveAttrs` when initializing a component with `factoryFor` has changed (see https://github.com/emberjs/ember.js/issues/17534 ). We run into this problem because a part of our components was relying on this.

To me it looks, like this hook should be called manually now, to preserve the normal component lifecycle. Maybe extend add the `trigger('didReceiveAttrs')` to the following code in `get-render-component.js`?

```js
// ...
let componentInstance = component.create(attrs);

componentInstance.trigger('didReceiveAttrs');
componentInstance.appendTo(element);

return componentInstance;
// ...
```

**Comment by mitchlloyd** [comment]
Thanks for reporting, Patric. If this behavior changed I wonder if there is
some way to apply a fix that wouldn’t potentially break past versions of
ember by invoking didReceiveAttrs twice.

**Comment by patric-eberle** [comment]
@mitchlloyd true. I'm not that familiar with Ember anymore unfortunately... The only thing that came to my mind is checking the current Ember version? Not very nice, but it would work I guess.

**Comment by mitchlloyd** [comment]
Maybe @rwjblue could advise a bit? I looks like we could inspect the Ember version to fix this bug (>= 3.2.0) and then [trigger the `didReceiveAttrs` hook](https://github.com/emberjs/ember.js/blob/2cd6e2f6d570d2489322e317adcb52a0756372d3/packages/ember-glimmer/lib/component-managers/curly.ts#L239), but I'm curious if there is a more official way. Perhaps I could use the component manager directly?

**Comment by rwjblue** [comment]
Yeah, I think detecting Ember version is the best bet. I don't think we expose the managers themselves (though that is a good idea and would help libs like this to avoid having to care about these specific nuances).

**Comment by xtagon** [comment]
Is there a solution or workaround for this? Does ember-islands support Ember 3.x?

Thanks in advance 👍
